crates/subspace-farmer/src/single_disk_farm/piece_cache.rs

```diff
@@ -81,6 +81,25 @@ impl farm::PieceCache for SingleDiskPieceCache {
             Ok(None)
         }
     }
+
+    async fn read_pieces(
+        &self,
+        offsets: Box<dyn Iterator<Item = PieceCacheOffset> + Send>,
+    ) -> Result<
+        Box<
+            dyn Stream<Item = Result<(PieceCacheOffset, Option<(PieceIndex, Piece)>), FarmError>>
+                + Send
+                + Unpin
+                + '_,
+        >,
+        FarmError,
+    > {
+        if let Some(piece_cache) = &self.maybe_piece_cache {
+            farm::PieceCache::read_pieces(piece_cache, offsets).await
+        } else {
+            Ok(Box::new(stream::empty()))
```

Comment:
Do we need to return `Ok((PieceCacheOffset, None))` for each offset in `offsets` here?

The documented invariant for the `farm::PieceCache read_pieces()` method is:
> Number of elements in returned stream is the same as number of unique `offsets`.